File: VB/MyGridCtrl/MyGridCtrl.vb
Imports Microsoft.VisualBasic
Imports DevExpress.XtraGrid
Imports System
Imports System.Collections.Generic
Imports System.Linq

Namespace DevExpress.S170736
	Public Class MyGridCtrl
		Inherits GridControl
		Protected Overrides Sub RegisterAvailableViewsCore(ByVal collection As XtraGrid.Registrator.InfoCollection)
			MyBase.RegisterAvailableViewsCore(collection)
			collection.Add(New MyGridInfoRegistrator())
		End Sub
		Protected Overrides Function CreateDefaultView() As XtraGrid.Views.Base.BaseView
			Return CreateView("MyBandedGridView")
		End Function
	End Class
End Namespace

File: CS/MyGridCtrl/MyGridCtrl.cs
using DevExpress.XtraGrid;
using System;
using System.Collections.Generic;
using System.Linq;

namespace DevExpress.S170736 {
    public class MyGridCtrl : GridControl {
        protected override void RegisterAvailableViewsCore(XtraGrid.Registrator.InfoCollection collection) {
            base.RegisterAvailableViewsCore(collection);
            collection.Add(new MyGridInfoRegistrator());
        }
        protected override XtraGrid.Views.Base.BaseView CreateDefaultView() {
            return CreateView("MyBandedGridView");
        }
    }
}
